drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -1114,6 +1157,30 @@ local function info_changed(driver, device, event, args)
       device:set_field(DEFERRED_CONFIGURE, nil)
     end
   end
+
+  if not device.preferences or device.network_type == device_lib.NETWORK_TYPE_CHILD then
+    return
+  end
+
+  if device.manufacturer_info.vendor_id == fingerprint_profile_overrides[1].vendor_id and
```

Comment:
This code is somewhat confusing and I think it could use some comments or documentation somewhere since there is some device specific knowledge here. For example, why is the time diff here and what is the significance of the values?

Reply:
I believe this code is used to prevent preferences from being changed faster than every two seconds. This was the same way that Inovelli implemented this for the zigbee driver, so it could be a device limitation. I will add some comments to make things a little more clear.